stdlib/messaging/rabbitmq/src/main/ballerina/rabbitmq/channel.bal

```diff
@@ -0,0 +1,132 @@
+    #
+    # + connection - Ballerina RabbitMQ Connection object.
+    extern function createChannel(Connection? connection);
+
+    # Closes this channel.
+    #
+    # + closeCode - The close code (See under "Reply Codes" in the AMQP specification).
+    # + closeMessage - A message indicating the reason for closing the connection.
+    public extern function close(int? closeCode, string? closeMessage) returns error?;
+
+    # Actively declare a server-named exclusive, autodelete, non-durable queue or queue with the given configurations.
+    #
+    # + config - Holds the paramters required to declare a queue.
+    # + return - Returns the name of the queue if autogenerated or nil if the queue was successfully generated with the
+    #               given parameters. An error is returned if an I/O error is encountered.
+    public remote extern function queueDeclare(QueueConfiguration? queueConfig = ()) returns string | error?;
+
+    # Actively declare a non-autodelete, non-durable exchange with no extra arguments, If the arguments are specifed,
+    # then the exchange is declared accordingly.
+    #
+    # + config - Holds parameters required to declare an exchange.
+    # + return - Returns an error if an I/O error is encountered or nil if successful.
+    public remote extern function exchangeDeclare(ExchangeConfiguration config) returns error?;
+
+    # Binds a queue to an exchange with the given binding key.
+    #
+    # + queueName - Name of the queue.
+    # + exchangeName - Name of the exchange.
+    # + bindingKey - Binding key used to bind the queue to the exchange.
+    # + return - Returns an error if an I/O error is encountered or nil if successful.
+    public remote extern function queueBind(string queueName, string exchangeName, string bindingKey) returns error?;
+
+    # Publishes a message. Publishing to a non-existent exchange will result in a channel-level
+    # protocol exception, which closes the channel.
+    #
+    # + routingKey - The routing key.
+    # + message - The message body.
+    # + exchange - The name of the exchange the message is published to.
+    # + return - Returns an error if an I/O error is encountered or nil if successful.
+    public remote extern function basicPublish(string routingKey = "", string exchange = "", string | byte[] message)
+                                      returns error?;
+
+    # Deletes a queue, without regard for whether it is in use or has messages on it, If the paramters ifUnused or
+    # ifEmpty is given, the queue is checked before deleting.
+    #
+    # + queueName - Name of the queue to be deleted.
+    # + ifUnused - True if the queue should be deleted only if not in use.
+    # + ifEmpty -  True if the queue should be deleted only if empty.
+    # + return - Returns error if an I/O error is encountered or nil if successful.
+    public remote extern function queueDelete(string queueName, boolean? ifUnused = false, boolean? ifEmpty = false)
```

Comment:
Why do we have to make `ifUnused` and `ifEmpty` optional types?

What do we do if the user sets `()` as the value for one of these?

Reply:
Same for `queueDelete` too.